lib/vrl/compiler/src/expression/if_statement.rs

```diff
@@ -17,11 +17,9 @@ pub struct IfStatement {
 
 impl IfStatement {
     pub(crate) fn noop() -> Self {
-        let literal = Literal::Boolean(false);
-        let predicate = Predicate::new_unchecked(vec![Expr::Literal(literal)]);
+        let predicate = Predicate::new_unchecked(vec![]);
 
-        let literal = Literal::Null;
-        let consequent = Block::new(vec![Expr::Literal(literal)], LocalEnv::default());
+        let consequent = Block::new(vec![Expr::Noop(Noop)], LocalEnv::default());
```

Comment:
These are the same. I decided against moving `Noop` behind a feature flag, as it's the simplest expression, and always resolves to `null`, which is useful in cases like these.

Reply:
Why is `vec![Expr::Noop(Noop)]` needed here rather than `vec![]`?

Reply:
Good catch, it technically isn't. I'll remove it.

Reply:
To clarify a bit on where the confusion comes from:

Blocks _must_ have expressions in them, otherwise they end up as empty objects:

```coffee
{} # empty object
{ null } # block
```

But, this difference is made at the parser level, so even though technically, it's impossible to have a block without expressions, nothing breaks if we do that here, since it's all guarded against using an `unwrap_or(Value::Null)` unwrap.

Having written this down, I think I prefer the clarity of injecting one expression here, in case we ever decide to reject empty blocks in the compiler as well.